### PR TITLE
Fix https://github.com/sonic-net/sonic-gnmi/issues/266

### PR DESCRIPTION
**Pull request by slicking**
Prevent deadlocking on the ChildSchemaMutex by ensuring no other go routine is holding the lock while the Entry struct is copied.

### CODE REVIEW

**Comment by adyeung** [comment]
@anand-kumar-subramanian pls review

**Review by ayelrod — APPROVED** [review]

File: patches/ygot/ygot.patch [hunk 1]
```diff
@@ -1627,7 +1627,24 @@ index ec52ee3..c312ec3 100644
...
+ 	// with ListAttrs unset.
++	// Since we've patched a mutex into the Entry struct we need special handling
++	// when making this copy.  We don't want to copy the mutex while it is locked
++	// by another goroutine, doing so would leave newSchema's copy of the mutex
++	// also locked but it would never be unlocked making it a deadlock risk.
++	// To avoid this, we lock the original mutex, make the copy, then unlock both
++	// the original and copy mutex.
++	schema.ChildSchemaMutex.Lock()
+ 	newSchema := *schema
++	schema.ChildSchemaMutex.Unlock()
```
Comment (mbalachandar) [inline patches/ygot/ygot.patch:1642]: please make it a common utility function to perform the copy schema or keep it as member function of the *yang.Entry

Comment (slicking) [reply patches/ygot/ygot.patch]: Updated as requested

**Comment by slicking** [comment]
@mbalachandar while this patch does prevent go routines from deadlocking on a bad mutex, it doesn't fully solve the problem.  The copy of the Entry struct results in the two Entry objects sharing a single instance of the ChildSchema map but using two different Mutex objects to protect access to it.  This can lead to "concurrent map read and map write" errors.

I will change this patch to move move all these objects (ChildSchema, ChildSchemaMutex, IsSchemaValidated) into a new struct, then add a pointer to that struct into the Entry object, and then ensure each Entry created initializes this struct.  The copy operation on the Entry operation will then result in both Entry objects sharing the same ChildSchema map and mutex which is safe.

This change looks simple enough, but since these errors are rare timing events I'll let it run in our tests for a few days.  If that doesn't show any problems I'll update this PR, probably towards end of this week.  If you feel there is a better way to address this issue than switching to a pointer in the Entry struct feel free to let me know.

**Review by mbalachandar — APPROVED** [review]

**Review by sneelam20 — APPROVED** [review]

**Pull request merged**